mobile/src/modules/home/components/HomeStackedAreaChart.tsx

```diff
@@ -178,14 +192,22 @@ export class HomeStackedAreaChart extends React.PureComponent<TProps, State> {
         />
 
         <TouchableOpacity
-          onPress={this.toggleBTCPrice}
+          onPress={this.selectCurrency}
           style={styles.button as StyleMedia}
         >
           <Text style={{ color: Colors.BLUE_LIGHT, textAlign: 'center' }}>
-            {this.state.priceInBTC ? `BTC` : `BURST`} TEST
+            {this.props.priceApi.selectedCurrency}
           </Text>
         </TouchableOpacity>
       </>
     );
   }
 }
+
+function mapStateToProps (state: State) {
+  return {
+    selectedDateRange: state.selectedDateRange
+  };
+}
+
+export const HomeStackedAreaChart = connect(mapStateToProps)(withNavigation(HomeStackedAreaChartComponent));
```

Comment:
had to do this to get access to the dispatcher in order to send the selectCurrency action to the reducer.

Reply:
nah, you can pass selected currency and `onCurrencyChange` via props
(also we don't need `withNavigation`)

use it like

```
onPress={this.props.onCurrencyChange}
```

and in parent

```
handleCurrencyChange = () => { ... }

...

<HomeStackedAreaChart onCurrencyChange={this.handleCurrencyChange} selectedCurrency={this.props.selectedCurrency} selectedDateRange={...} ...
```

Reply:
I also need access to other parts of the `priceApi`, and it didn't feel right breaking that data up into multiple props when I should be able to access them via this.props already.